tests/test_settings.py

```diff
@@ -79,6 +79,18 @@ def test_unknown_setting(self):
         settings = {'unknown_setting': 100500}
         self.client.execute('SELECT 1', settings=settings)
 
+    # DBMS_MIN_REVISION_WITH_SETTINGS_SERIALIZED_AS_STRINGS is 20.1.2+
+    @require_server_version(20, 1, 2)
+    def test_unknown_setting_is_important(self):
+        # In case of rev >= DBMS_MIN_REVISION_WITH_SETTINGS_SERIALIZED_AS_STRINGS
+        # and setting marked as important, then the query should fail.
+        settings = {'unknown_setting': 100500}
+        self.client.connection.settings_is_important = True
```

Comment:
Context manager `self.created_client` helps with patching internals.

```python
            with self.created_client(settings_is_important=True) as client:
                client.execute('SELECT 1', settings=settings)
```

Reply:
Thanks! (This was already pops up in my head)